src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,109 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [description, setDescription] = useState('');
+  const [errors, setErrors] = useState({
+    title: false,
+    imgUrl: false,
+    imdbUrl: false,
+    imdbId: false,
+  });
```

Comment:
You don't need that much hooks just use one which stores an object. And you don't need to create state for errors with object in it. Actually you have already 'required' in the input elements and it works so your implementation duplicates code and error message, just check if required field are not empty and according to that make it disabled or not. 
```suggestion
export const NewMovie: React.FC<Props> = ({ onAdd }) => {
  const [newFilm, setNewFilm] = useState<FilmType | null>(null);
```